isTestServerTrust() in ServiceWorkerBasic.mm should use WTF::bridge_cast<>

ddkilzer opened the pull request:
#### 90640da0da75182bbe5fce97a1f98cd6d0e3fc67
<pre>
isTestServerTrust() in ServiceWorkerBasic.mm should use WTF::bridge_cast&lt;&gt;
&lt;<a href="https://bugs.webkit.org/show_bug.cgi?id=274659">https://bugs.webkit.org/show_bug.cgi?id=274659</a>&gt;
&lt;<a href="https://rdar.apple.com/128686657">rdar://128686657</a>&gt;

Reviewed by Darin Adler.

* Tools/TestWebKitAPI/Tests/WebKitCocoa/ServiceWorkerBasic.mm:
- Switch from adoptNS() to adoptCF().get(), and use
  WTF::bridge_cast&lt;&gt;() instead of (NSString *) to toll-free bridge the
  CFStringRef to NSString.

Canonical link: <a href="https://commits.webkit.org/279307@main">https://commits.webkit.org/279307@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/192e5aff98148045f8a8d3cbc9b2fc1be5acb980

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/53024 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/32361 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/5511 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/56303 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/59/builds/3747 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/55329 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/39251 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/3473 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/43010 "Passed tests") | [✅ 🧪 wincairo-tests](https://ews-build.webkit.org/#/builders/60/builds/2424 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/55122 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/30160 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/45771 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/24137 "Passed tests") | 
| | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/42/builds/27161 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/3091 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/1906 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/49013 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/3248 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/57898 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/28165 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/62/builds/3211 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/50403 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/29385 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/45988 "Passed tests") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/21/builds/49710 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/30304 "Built successfully") | | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/7803 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/29139 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

darinadler commented on Tools/TestWebKitAPI/Tests/WebKitCocoa/ServiceWorkerBasic.mm:
```diff
@@ -2471,7 +2472,7 @@ static bool isTestServerTrust(SecTrustRef trust)
     auto chain = adoptCF(SecTrustCopyCertificateChain(trust));
     auto certificate = checked_cf_cast<SecCertificateRef>(CFArrayGetValueAtIndex(chain.get(), 0));
-    if (![adoptNS((NSString *)SecCertificateCopySubjectSummary(certificate)) isEqualToString:@"Me"])
+    if (![adoptNS(bridge_cast(SecCertificateCopySubjectSummary(certificate))) isEqualToString:@"Me"])
```
Seems like we must do an `adoptCF` here, not an `adoptNS`. Then we can do `bridge_cast` on the result of `get()`.

ddkilzer replied:
Yeah, that's a better option in the long run.

Commit:
isTestServerTrust() in ServiceWorkerBasic.mm should use WTF::bridge_cast<>
<https://bugs.webkit.org/show_bug.cgi?id=274659>
<rdar://128686657>

Reviewed by Darin Adler.

* Tools/TestWebKitAPI/Tests/WebKitCocoa/ServiceWorkerBasic.mm:
- Switch from adoptNS() to adoptCF().get(), and use
  WTF::bridge_cast<>() instead of (NSString *) to toll-free bridge the
  CFStringRef to NSString.

Canonical link: https://commits.webkit.org/279307@main
